src/main/java/io/confluent/idesidecar/restapi/cache/ClusterCache.java

```diff
@@ -191,6 +191,9 @@ void onConnectionCreated(@ObservesAsync @Lifecycle.Created ConnectionState conne
 
   void onConnectionUpdated(@ObservesAsync @Lifecycle.Updated ConnectionState connection) {
     Log.infof("Updated %s", connection.getSpec());
+
+    // clear the cache to force future Kafka/SR REST proxy requests to fetch and re-cache any new cluster details
+    clustersByConnectionId.remove(connection.getSpec().id());
```

Comment:
If we just remove it, then when the cache is used to find clusters for a particular connection, that connection will not be found and we'll get an exception.

I think we actually want to _replace_ the connection's cache:
```suggestion

    // Replace the existing cache for this connection with a new one
    clustersByConnectionId.put(
        connection.getId(),
        new Clusters(connection.getId(), connection.getType())
    );
```
BTW, we could also have changed `Clusters` to have a `clear()` method, but that's not as good as _replacing_ the `Clusters` object for the connection. Why? The `Clusters` class has two maps, and it's not possible to atomically clear both. (Another thread might see an inconsistent `Clusters` object between our clearing of the two maps.) However, `clustersByConnectionId` is a concurrent map, and replacing the existing value on that entry requires changing one reference, which is atomic.